### PR TITLE
wrong cache after create while active_test=False

### PR DESCRIPTION
**Pull request by seb-odoo**
Introduce failing test to highly the issue.

The problem seems to be that the new values are added to the cache depending on
active_test and not depending on the actual value of the active field.

--------

The original issue is that installation of themes is broken, method `_convert_to_base_model` in `theme_models.py`.

The line
`inherit = self.inherit_id.with_context(active_test=False).copy_ids.filtered(lambda x: x.website_id == website)`
is returning an empty set because of `active_test=False`, even though values actually exist (when doing the same line with `active_test=True`).

To reproduce: install any theme on a website through the settings menu, add a breakpoint in that method inside `if not inherit:` add `if self.inherit_id.copy_ids:`, that condition should never trigger but it does.

----

Note: tests here seem to fail on `active_test=True` while the theme issue is on `active_test=False` so I'm not 100% sure that the tests here cover all problematic cases.

### CODE REVIEW

File: odoo/addons/test_new_api/tests/test_new_fields.py [hunk 1]
```diff
@@ -1539,6 +1539,13 @@ def test_11_ondelete_many2many(self):
         record_a.unlink()
         self.assertFalse(record_a.exists())
 
+    def test_12_active_test_one2many(self):
+        parent = self.env['test_new_api.model_active_field'].create({})
+        self.assertFalse(parent.children_ids)
+        child = self.env['test_new_api.model_active_field'].create({'multi': parent.id})
+        self.assertEqual(parent.children_ids, child)
+        self.assertEqual(parent.with_context(active_test=False).children_ids, child)
```
Comment (seb-odoo) [inline odoo/addons/test_new_api/tests/test_new_fields.py]: The problem is here, `children_ids` is empty.

File: odoo/addons/test_new_api/tests/test_new_fields.py [hunk 2]
```diff
@@ -1539,6 +1539,20 @@ def test_11_ondelete_many2many(self):
...
+        self.assertFalse(parent.children_ids)
+
+        # create with active_test=True
+        child1 = self.env['test_new_api.model_active_field'].create({'parent_id': parent.id})
+        self.assertEqual(parent.children_ids, child1)
+        self.assertEqual(parent.with_context(active_test=False).children_ids, child1)
+
+        # create with active_test=False
+        child2 = self.env['test_new_api.model_active_field'].with_context(active_test=False).create({'parent_id': parent.id})
+        self.assertEqual(parent.children_ids, child1 + child2)
```
Comment (seb-odoo) [inline odoo/addons/test_new_api/tests/test_new_fields.py]: The problem is that here child2 is not added to cache so it crashes.

**Comment by seb-odoo** [comment]
cc @qsm-odoo 

Not sure if/how to fix at ORM level but adding `self.inherit_id.invalidate_cache(fnames=['copy_ids'])` before the failing line seems to do the trick.

**Comment by rco-odoo** [comment]
@beledouxdenis any idea?

**Comment by rco-odoo** [comment]
Found a better fix in #37577.

**Pull request closed without merging**